src/test/java/com/yelp/nrtsearch/server/grpc/TimeoutTest.java

```diff
@@ -0,0 +1,399 @@
+            .setQuery(getQuery())
+            .setTimeoutSec(1000)
+            .build();
+
+    verifyNoTimeout(request);
+  }
+
+  @Test
+  public void testFacetQueryNoTimeout() throws Exception {
+    SearchRequest request =
+        SearchRequest.newBuilder()
+            .setIndexName(TEST_INDEX)
+            .setTopHits(NUM_DOCS)
+            .addAllRetrieveFields(RETRIEVE_LIST)
+            .setQuery(getQuery())
+            .setTimeoutSec(1000)
+            .addFacets(getTestFacet())
+            .build();
+
+    verifyNoTimeout(request);
+  }
+
+  private void verifyNoTimeout(SearchRequest request) throws Exception {
+    TopDocs hits =
+        queryWithFunction(
+            request,
+            context -> {
+              try {
+                TopDocs topDocs = getTopDocs(context, context.getCollector().getWrappedManager());
+                assertFalse(context.getCollector().hadTimeout());
+                return topDocs;
+              } catch (IOException e) {
+                throw new RuntimeException(e);
+              }
+            });
+    assertEquals(NUM_DOCS, hits.totalHits.value);
+    assertEquals(NUM_DOCS - 1, hits.scoreDocs[0].score, 0);
+  }
+
+  @Test
+  public void testQueryPartialResults() throws Exception {
+    SearchRequest request =
+        SearchRequest.newBuilder()
+            .setIndexName(TEST_INDEX)
+            .setTopHits(NUM_DOCS)
+            .addAllRetrieveFields(RETRIEVE_LIST)
+            .setQuery(getQuery())
+            .setTimeoutSec(5)
+            .build();
+
+    verifyPartialResults(request);
+  }
+
+  @Test
+  public void testFacetQueryPartialResults() throws Exception {
+    SearchRequest request =
+        SearchRequest.newBuilder()
+            .setIndexName(TEST_INDEX)
+            .setTopHits(NUM_DOCS)
+            .addAllRetrieveFields(RETRIEVE_LIST)
+            .setQuery(getQuery())
+            .setTimeoutSec(5)
+            .addFacets(getTestFacet())
+            .build();
+
+    verifyPartialResults(request);
+  }
+
+  private void verifyPartialResults(SearchRequest request) throws Exception {
+    TopDocs hits =
+        queryWithFunction(
+            request,
+            context -> {
+              boolean[] hasTimeout = new boolean[1];
+              try {
+                TopDocs topDocs =
+                    getTopDocs(
+                        context,
+                        new TimeoutWrapper(
+                            (CollectorManager<Collector, TopDocs>)
+                                context.getCollector().getManager(),
+                            5,
+                            false,
+                            () -> hasTimeout[0] = true));
+                assertTrue(hasTimeout[0]);
+                return topDocs;
+              } catch (IOException e) {
+                throw new RuntimeException(e);
+              }
+            });
+    assertEquals(20, hits.totalHits.value);
+  }
+
+  @Test(expected = CollectionTimeoutException.class)
```

Comment:
Will this exception be thrown even though `disallowPartialResults` is not set in the request ? Same for the next test too

Reply:
The test is using its own version of the timeout wrapper, so it doesn't need to configure the one added by the request processor.